int32 part
int32 status
---
int32 res